extensions/stackdriver/stackdriver.cc

```diff
@@ -94,27 +103,33 @@ bool StackdriverRootContext::onConfigure(
 }
 
 void StackdriverRootContext::onStart(std::unique_ptr<WasmData>) {
-#ifndef NULL_PLUGIN
-// TODO: Start a timer to trigger exporting
-#endif
+  if (!config_.disable_access_logging()) {
+    proxy_setTickPeriodMilliseconds(kDefaultLogExportMilliseconds);
+  }
 }
 
 void StackdriverRootContext::onTick() {
-#ifndef NULL_PLUGIN
-// TODO: Add exporting logic with WASM gRPC API
-#endif
+  if (!config_.disable_access_logging()) {
+    logger_->exportLogEntry();
+  }
 }
 
 void StackdriverRootContext::record(const RequestInfo &request_info,
                                     const NodeInfo &peer_node_info) {
   ::Extensions::Stackdriver::Metric::record(isOutbound(), local_node_info_,
                                             peer_node_info, request_info);
+  if (!config_.disable_access_logging()) {
+    logger_->addLogEntry(request_info, peer_node_info);
+  }
 }
 
 inline bool StackdriverRootContext::isOutbound() {
   return direction_ == ::Wasm::Common::TrafficDirection::Outbound;
 }
 
+// TODO(bianpengyuan) Add final export once root context supports onDone.
+// https://github.com/envoyproxy/envoy-wasm/issues/240
```

Comment:
Can this be added to contextdestructor ?

Reply:
No, wasm runtime assumes callbacks from modules into hosts must be triggered inside a host call into module (e.g. onDone), otherwise it does not have context information about the callback.